runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3063,21 +3063,25 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 		result = false;
 	}
 
-	/* Init of thread count will only occur if it is not enforced by gcParseReconfigurableArguments. */
-	extensions->configuration->initializeGCThreadCount(env);
-
 	/* If the thread count is being forced, check its validity and display a warning message if it is invalid, then mark it as invalid (see comment below). */
 	if (extensions->gcThreadCountForced && (extensions->gcThreadCount < extensions->dispatcher->threadCountMaximum())) {
 		j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_GC_THREAD_VALUE_MUST_BE_ABOVE_WARN, (UDATA)extensions->dispatcher->threadCountMaximum());
 		extensions->gcThreadCountForced = false;
 	}
 
+	/* Init of thread count can only occur if not enforced by gcParseReconfigurableArguments.
+	 * This must be done after validation of a forced thread count since a new deault thread count may need to be
+	 * computed in case the forced thread count is invalid. */
+	extensions->configuration->initializeGCThreadCount(env);
+
 	/* Currently, threads don't shutdown during restore, so ensure thread count doesn't fall below
 	 * the checkpoint thread count. This adjustment can be removed in the future when dispatcher
 	 * thread shutdown is sufficiently tested at restore.
 	 */
 	extensions->gcThreadCount = OMR_MAX(extensions->dispatcher->threadCountMaximum(), extensions->gcThreadCount);
 
+	extensions->configuration->initializeGCParameters(env);
```

Comment:
consider converging the last few lines into just one line: configuration->reinitialize()

Reply:
Moved to Configuration's reinitializeForRestore: https://github.com/eclipse/omr/pull/7035

Reply:
That's good, but could other 2 lines above it also go there?
The point is there should be only one invocation against Configuration (and that is reinitializeForRestore()) and then that one should call other Configuration methods if needed...